image-rs/src/signature/mechanism/simple/mod.rs

```diff
@@ -177,27 +234,142 @@ pub fn judge_single_signature(
     Ok(())
 }
 
+#[cfg(feature = "signature-simple-xrss")]
+impl RegistryClient {
+    pub fn new() -> RegistryClient {
+        RegistryClient {
+            client: reqwest::Client::new(),
+        }
+    }
+
+    pub async fn get_oauth_token_for_registry(
+        &self,
+        image: &Image,
+        auth: &RegistryAuth,
+    ) -> Result<String> {
+        match auth {
+            RegistryAuth::Anonymous => {
+                bail!("Trying to get signature from a registry without providing auth information")
+            }
+            RegistryAuth::Basic(username, password) => {
+                let scope = format!("repository:{}:pull", image.reference.repository());
+                let res = self
+                    .client
+                    .get(format!(
+                        "https://{}/oauth/token",
+                        image.reference.registry()
+                    ))
+                    .query(&[
+                        ("account", username),
+                        ("scope", &scope),
+                        ("service", &"registry".to_string()),
+                    ])
+                    .basic_auth(username, Some(password))
+                    .send()
+                    .await
+                    .map_err(|e| anyhow!("Failed to fetch oauth token for registry: {:?}", e))?;
+
+                let oauth_token_response_body: GetOAuthTokenResponse =
+                    res.json::<GetOAuthTokenResponse>().await.map_err(|e| {
+                        anyhow!(
+                            "Unexpected response from fetching oauth token from registry: {:?}",
+                            e
+                        )
+                    })?;
+
+                Ok(oauth_token_response_body.token)
+            }
+        }
+    }
+
+    pub async fn get_signatures_from_registry(
```

Comment:
maybe we can send the PR in parallel to https://github.com/krustlet/oci-distribution to extend the `Client` API to support pull signatures from registry